x/supplier/types/message_stake_supplier.go

```diff
@@ -12,21 +12,35 @@ const TypeMsgStakeSupplier = "stake_supplier"
 var _ sdk.Msg = (*MsgStakeSupplier)(nil)
 
 func NewMsgStakeSupplier(
-	address string,
+	signerAddress string,
+	ownerAddress string,
+	supplierAddress string,
 	stake sdk.Coin,
 	services []*sharedtypes.SupplierServiceConfig,
 ) *MsgStakeSupplier {
 	return &MsgStakeSupplier{
-		Address:  address,
-		Stake:    &stake,
-		Services: services,
+		Signer:       signerAddress,
+		OwnerAddress: ownerAddress,
+		Address:      supplierAddress,
+		Stake:        &stake,
+		Services:     services,
 	}
 }
 
 func (msg *MsgStakeSupplier) ValidateBasic() error {
```

Comment:
Does it make sense to move the check which ensures that the signer is either the owner or operator here instead?

Reply:
We can't, since we may have an external signer (neither the operator, nor the owner) that can stake the funds.

See: https://discord.com/channels/553741558869131266/1266860790917632121/1271475296805322762